### ISSUE TITLE
'Bad Authentication data' exception

### ISSUE THREAD

**Issue by tmetro** [semifor/Twitter-API#13]
I have some code that calls user_timeline() once an hour in a loop. After 37 polls, it failed with a 'Bad Authentication data' exception.

The method call was wrapped in a try{} block, with the catch handler including:
```
die $_ unless Twitter::API::Util->is_twitter_api_error;
say "Twitter error: ",dump($_);
```

The resulting error message was a simple string, implying is_twitter_api_error() returned false. Is authentication not considered part of the Twitter API?

My recollection of OAuth is that the client uses the access token to obtains a session token, which can expire. Does your implementation have handling to refresh that session token when it has expired?

**Comment by hoelzro** [comment]
I encountered the same behavior - I found the issue and created a [PR](#14)

**Comment by semifor** [comment]
Fixed in #14.

**Comment by tmetro** [comment]
I just wanted to confirm that this is working in my testing.

After this patch was released, I reverted the workarounds I had in place, and restarted my bot. It's ran for multiple weeks using a single instance of the API object and is now apparently handling refreshing the authentication automatically.

Thanks @hoelzro for the patch.
